core/commands/destroy_command.rb

```diff
@@ -105,13 +105,27 @@ def destroy_by_configuration
     else
       vagrant_cleaner = VagrantCleaner.new(@env, @ui)
       vagrant_cleaner.destroy_nodes_by_configuration(configuration)
+      update_configuration_files
       return unless @env.labels.nil? && Configuration.config_directory?(@args.first)
 
       remove_files(configuration, @env.keep_template)
       vagrant_cleaner.destroy_aws_keypair(configuration)
     end
   end
 
+  # Update network_configuration and configured_labels files
+  def update_configuration_files
+    args = @args.first.split('/')
+    configuration = Configuration.new(args[0], @env.labels)
+    configurator = VagrantConfigurator.new(@specification, configuration, @env, @ui)
```

Comment:
By the name this class is responsible for creating the configuration, not deleting it. Direct usage of this class seems to be overkill.

You should have extracted the required functionality into another class and use it in both VagrantConfigurator and DestroyCommand classes.